arch/s390/dfltcc_common.c

```diff
@@ -0,0 +1,80 @@
+/* dfltcc_deflate.c - IBM Z DEFLATE CONVERSION CALL support. */
+
+#include "dfltcc_common.h"
+
+#include <stdio.h>
+#include <errno.h>
+#include <stddef.h>
+#include <stdlib.h>
+
+int env_dfltcc_disabled;
+int env_dfltcc_source_date_epoch;
+unsigned long env_dfltcc_level_mask;
+unsigned long env_dfltcc_block_size;
+unsigned long env_dfltcc_block_threshold;
+unsigned long env_dfltcc_dht_threshold;
+unsigned long env_dfltcc_ribm;
+
+static unsigned long xstrtoul(const char *s, unsigned long _default)
+{
+    char *endptr;
+    unsigned long result;
+
+    if (!(s && *s))
+        return _default;
+    errno = 0;
+    result = strtoul(s, &endptr, 0);
+    return (errno || *endptr) ? _default : result;
+}
+
+__attribute__((constructor)) static void init_globals(void)
+{
+    const char *env;
+
+    env = getenv("DFLTCC");
+    env_dfltcc_disabled = env && !strcmp(env, "0");
+
+    env = getenv("SOURCE_DATE_EPOCH");
+    env_dfltcc_source_date_epoch = !!env;
+
+#ifndef DFLTCC_LEVEL_MASK
+#define DFLTCC_LEVEL_MASK 0x2
+#endif
+    env_dfltcc_level_mask = xstrtoul(getenv("DFLTCC_LEVEL_MASK"),
+                              DFLTCC_LEVEL_MASK);
+    if(env_dfltcc_level_mask > 10) {
+        fprintf(stderr, "DFLTCC_LEVEL_MASK %lu out of range. Allowed range is 0-10. Resetting to default(%d)\n", 
```

Comment:
While having diagnostics is good, I don't think we should be printing to console from a widely used shared library. This may create nasty consequences if it's used in software which treats stderr as machine-readable.

Reply:
i see this options:

1. only print if `isatty() == 1) 
2. only print with special DEBUG compile flag
3. don't print at all

what would be an acceptable solution?

Reply:
There is a precedent with ZLIB_DEBUG in deflate_p.h, so I guess doing the same would be acceptable.

Reply:
Either 2 or 3 is fine for me...